doc/TESTS.rst

```diff
@@ -114,27 +114,41 @@ a test class::
           with pytest.raises(ValueError, match='.*some matching regex.*'):
               ...
 
-Within these test methods, ``assert`` and related functions are used to test
-whether a certain assumption is valid. If the assertion fails, the test fails.
+Within these test methods, the ``assert`` statement or a specialized assertion
+function is used to test whether a certain assumption is valid. If the
+assertion fails, the test fails. Common assertion functions include:
+
+- :func:`numpy.testing.assert_equal` for testing exact elementwise equality
+  between a result array and a reference,
+- :func:`numpy.testing.assert_allclose` for testing near elementwise equality
+  between a result array and a reference (i.e. with specified relative and
+  absolute tolerances), and
```

Comment:
Maybe we should advertise the `_nulp` versions more?

Reply:
Maybe. But actually, what are NumPy 2.0's plans for the zoo of testing functions? Can the offerings be pared down to those that are recommended?

Reply:
There is no such plan. The usage of older testing functions is too high to do that I'm afraid. The functions that were deprecated/removed typically had a usage count of a handful, in some cases up to ~30 or so, in SciPy, scikit-learn and Pandas. I just did a quick count for `assert_array_almost_equal` and it's 2838 in SciPy alone - so we can't touch it really.

This topic can be better addressed in the docs though. This PR is a great step in the right direction.

Reply:
OK. But how about listing `assert_array_equal` as "not recommended" following the plan suggested in https://github.com/numpy/numpy/issues/8457#issuecomment-1714230145?